src/DBmysql.php

```diff
@@ -349,6 +349,24 @@ public function escape($string)
      * @return mysqli_result|boolean Query result handler
      */
     public function query($query)
+    {
+        $e = new \RuntimeException('Executing direct queries is not allowed!');
+        var_export($e->getTraceAsString());
+        throw $e;
```

Comment:
If we want to fail here, following code will probably be handled more correctly by GLPI error handler.
```suggestion
        trigger_error('Executing direct queries is not allowed!', E_USER_ERROR);
```